tests/Unit/Helpers/Evolution/DiscontinuousGalerkin/BoundaryCorrections.hpp

```diff
@@ -0,0 +1,317 @@
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& unit_normal_covector,
+    const boost::optional<tnsr::I<DataVector, Dim, Frame::Inertial>>&
+        mesh_velocity) {
+  boost::optional<Scalar<DataVector>> normal_dot_mesh_velocity{};
+  if (static_cast<bool>(mesh_velocity)) {
+    normal_dot_mesh_velocity =
+        dot_product(*mesh_velocity, unit_normal_covector);
+  }
+  correction.dg_package_data(
+      make_not_null(&get<PackageTags>(*package_data))...,
+      get<FaceTags>(face_variables)..., unit_normal_covector, mesh_velocity,
+      normal_dot_mesh_velocity, get<VolumeTags>(volume_data)...);
+}
+
+template <typename BoundaryCorrection, typename... BoundaryCorrectionTags,
+          typename... PackageTags>
+void call_dg_boundary_terms(
+    const gsl::not_null<Variables<tmpl::list<BoundaryCorrectionTags...>>*>
+        boundary_corrections,
+    const BoundaryCorrection& correction,
+    const Variables<tmpl::list<PackageTags...>>& interior_package_data,
+    const Variables<tmpl::list<PackageTags...>>& exterior_package_data,
+    const ::dg::Formulation dg_formulation) {
+  correction.dg_boundary_terms(
+      make_not_null(&get<BoundaryCorrectionTags>(*boundary_corrections))...,
+      get<PackageTags>(interior_package_data)...,
+      get<PackageTags>(exterior_package_data)..., dg_formulation);
+}
+
+template <typename System, typename BoundaryCorrection, size_t FaceDim,
+          typename... VolumeTags>
+void test_boundary_correction_impl(
+    const BoundaryCorrection& correction, const Mesh<FaceDim>& face_mesh,
+    const tuples::TaggedTuple<VolumeTags...>& volume_data,
+    const bool use_moving_mesh, const ::dg::Formulation dg_formulation,
+    const ZeroOnSmoothSolution zero_on_smooth_solution) {
+  CAPTURE(use_moving_mesh);
+  CAPTURE(dg_formulation);
+  CAPTURE(FaceDim);
+  using variables_tag = typename System::variables_tag;
+  using variables_tags = typename variables_tag::tags_list;
+  using primitive_variables_tags = detail::get_system_primitive_vars<
+      System::has_primitive_and_conservative_vars, System>;
+  using flux_variables = typename System::flux_variables;
+  using flux_tags =
+      db::wrap_tags_in<::Tags::Flux, flux_variables, tmpl::size_t<FaceDim + 1>,
+                       Frame::Inertial>;
+  using temporary_tags =
+      typename System::compute_volume_time_derivative_terms::temporary_tags;
+  using dt_variables_tags = db::wrap_tags_in<::Tags::dt, variables_tags>;
+
+  using dg_package_field_tags =
+      typename BoundaryCorrection::dg_package_field_tags;
+  using package_temporary_tags =
+      typename BoundaryCorrection::dg_package_data_temporary_tags;
+  using package_primitive_tags = detail::get_correction_primitive_vars<
+      System::has_primitive_and_conservative_vars, BoundaryCorrection>;
+
+  // Check that the temporary tags needed on the boundary are listed as
+  // temporary tags for the volume time derivative computation.
```

Comment:
[clarity nitpick] Good comment, except the words used don't match the variable names. I think it is easy to fix, by adding the variable names in parentheticals, like this:

"Check that the temporary tags needed on the boundary (`package_temporary_tags`) are listed as temporary tags for the volume time derivative computation (`temporary_tags`)"

... or the opposite way around, if I've misunderstood.